File: quark/test/emit/defaulted-methods/py/defaulted_methods.py
from quark_runtime import *

import pkg



def main():
    t1 = pkg.T1();
    (t1).foo();
    (t1).bar();
    _println(u"===");
    t2 = pkg.T2();
    (t2).foo();
    (t2).bar();
    _println(u"===");
    t3 = pkg.T3();
    (t3).foo();
    (t3).bar();
    _println(u"===");
    t4 = pkg.T4();
    (t4).foo();
    (t4).bar();
    _println(u"===");
    t5 = pkg.T5();
    (t5).foo();
    (t5).bar();


if __name__ == "__main__":
    main()

File: quarkc/test/emit/expected/rb/defaulted-methods/lib/defaulted_methods.rb
module Quark
require "quark"
def self.defaulted_methods; DefaultedMethods; end
module DefaultedMethods
require_relative 'pkg' # 0 () ()
require_relative 'defaulted_methods' # 0 () ()

def self.call_main()
    self.main(::DatawireQuarkCore::List.new(ARGV))

    nil
end
def self.main(args)
    
    t1 = ::Quark.pkg.T1.new()
    t1.foo()
    t1.bar()
    ::DatawireQuarkCore.print("===")
    t2 = ::Quark.pkg.T2.new()
    t2.foo()
    t2.bar()
    ::DatawireQuarkCore.print("===")
    t3 = ::Quark.pkg.T3.new()
    t3.foo()
    t3.bar()
    ::DatawireQuarkCore.print("===")
    t4 = ::Quark.pkg.T4.new()
    t4.foo()
    t4.bar()
    ::DatawireQuarkCore.print("===")
    t5 = ::Quark.pkg.T5.new()
    t5.foo()
    t5.bar()


    nil
end

if __FILE__ == $0 then ::Quark.defaulted_methods.call_main() end

end # module DefaultedMethods
end # module Quark
